framework/src/media/MediaPlayerImpl.cpp

```diff
@@ -544,16 +544,46 @@ void MediaPlayerImpl::playback()
 	ssize_t num_read = mInputDataSource->read(mBuffer,(int)mBufSize);
 	meddbg("num_read : %d\n", num_read);
 	if (num_read > 0) {
-		start_audio_stream_out(mBuffer, get_output_bytes_to_frame((unsigned int)num_read));
+		int ret = start_audio_stream_out(mBuffer, get_output_bytes_to_frame((unsigned int)num_read));
+		if (ret < 0) {
+			PlayerWorker& mpw = PlayerWorker::getWorker();
+			switch (ret) {
+			case AUDIO_MANAGER_XRUN_STATE:
+				meddbg("AUDIO_MANAGER_XRUN_STATE\n");
+				break;
+			default:
+				meddbg("audio manager error : %d\n", ret);
+				notifyObserver(PLAYER_OBSERVER_COMMAND_ERROR);
+				mpw.enQueue(&MediaPlayerImpl::stopByError, shared_from_this());;
+				break;
+			}
+		}
 	} else if (num_read == 0) {
 		notifyObserver(PLAYER_OBSERVER_COMMAND_FINISHIED);
 		stop();
 	} else {
+		PlayerWorker& mpw = PlayerWorker::getWorker();
+		meddbg("InputDatasource read error\n");
 		notifyObserver(PLAYER_OBSERVER_COMMAND_ERROR);
-		stop();
+		mpw.enQueue(&MediaPlayerImpl::stopByError, shared_from_this());;
 	}
 }
 
+void MediaPlayerImpl::stopByError()
+{
+	PlayerWorker& mpw = PlayerWorker::getWorker();
+
+	meddbg("PlayerWorker Worker : stopOnPlaybackByError\n");
+
+	audio_manager_result_t result = stop_audio_stream_out();
```

Comment:
I forgot this thing. playerStop also needs stop_audio_stream_out.

Reply:
stopPlayer already call stop_audio_stream_out.
https://github.com/Samsung/TizenRT/commit/f27ddd23da07773d5293862bb132dab86261a9a1